inc/supplier.class.php

```diff
@@ -459,26 +459,32 @@ function showInfocoms() {
                $linkfield = 'consumableitems_id';
             }
 
+            if ($itemtype == 'Item_DeviceControl') {
+               $query .= "INNER JOIN `glpi_devicecontrols`
+                           ON (`glpi_items_devicecontrols`.`devicecontrols_id`=`glpi_devicecontrols`.`id`)";
+               $linktype = 'DeviceControl';
+               $linkfield = 'devicecontrols_id';
+            }
+
             $linktable = getTableForItemType($linktype);
+            $link_item = new $linktype();
```

Comment:
This code was already existing. It was used to call `getFormURLWithID()` which is static also. WIll do the change.